### DISCUSSION TITLE
Compatibility gcc9 and gcc10

### DISCUSSION THREAD

**Discussion by mryzhov** [pybind/pybind11#5418] [General]
I build common libraries(https://pypi.org/project/openvino/) using the manylinux image (2014, GCC 10)  and then build a dependent module (https://pypi.org/project/openvino-genai/) that uses types from the common module. The compilation with GCC 9 on Ubuntu 20 is successful, but when I call the methods from the second module, it cannot find the types wrapped in the first module. Both are built with the -D_GLIBCXX_USE_CXX11_ABI=1 flag.

The error I got: "TypeError: Unregistered type : ov::Tensor", where ov::Tensor is a class defined in the common module.

Could you please advise me on how I could properly build both modules to be compatible with each other?

**Comment by ilya-lavrenov** [comment]
The same question for MSVC 2019 and 2022, which cannot be fixed.
Could you, pybind11 developers, explain why?

is there anyway to fix it or at least provide better error message why "child" module does not see types defined in "parent" module?

**Reply by shjiyang-intel** [reply]
Which version of MSVC do not have this issue? Or have you find out other fix for it? thank you
